src/java/PermutationManager.java

```diff
@@ -14,39 +14,41 @@
 public class PermutationManager
 {
 
-    private static PermutationManager singleton;
-    private static Integer lock;
+    private static PermutationManager singleton; // declares that there can only ever be one PermutationManager instance at a time?
+    private static Integer lock; // declared for the purposes for the synchronized block later?
 
-    public List<Peer> peers;
-    public HashMap<Peer, Integer> peerIndexHashMap;
-    public HashMap<Sha256Hash, TransactionPermutation> transactionPermutationHashMap;
+    public List<Peer> peers; //array of peer objects
+    public HashMap<Peer, Integer> peerIndexHashMap; //hashmap of peer to index relationship
+    public HashMap<Sha256Hash, TransactionPermutation> transactionPermutationHashMap; //hashmap of the transaction's hash and a permutation object
+    // which contains a list of integers corresponding to the peer/index relationship in peerIndexHashMap
 
     public PermutationManager(List<Peer> peers)
     {
+        //constructor
         this.peers = peers;
         this.peerIndexHashMap = new HashMap<Peer, Integer>();
         this.transactionPermutationHashMap = new HashMap<Sha256Hash, TransactionPermutation>();
 
         int i = 0;
         for (Peer peer : peers)
         {
-            peerIndexHashMap.put(peer, i++);
+            peerIndexHashMap.put(peer, i++); // establishes the peer/index relationship in the peerIndexHashMap
         }
     }
 
     public static PermutationManager getPermutationManager()
     {
-        return singleton;
+        return singleton; //calls the constructor as a singleton?
```

Comment:
This should create the singleton if it doesn't exist. This is fixed in my next PR. But this line, specifically, doesn't call the constructor, but it should 👍